src/porepy/numerics/mixed_dim/assembler.py

```diff
@@ -1202,3 +1204,120 @@ def num_dof(self):
+                if len(var) != len(var.intersection(set(unique_vars))):
+
+                    # We will only report each subset variable definition once.
+                    # By default, assume this subset is new
+                    new_comb = True
+                    # Check if we have found it before
+                    for spec in found_special_var_combination:
+                        if len(spec) == len(spec.intersection(set(var))):
+                            new_comb = False
+                            break
+                    # if not bound before, register the new one, and report it
+                    if new_comb:
+                        found_special_var_combination.append(var)
+                        s += "Variable subset on at least one subdomain in "
+                        s += " dimension {dim}: {var}\n"
+
+        for dim in range(self.gb.dim_max(), self.gb.dim_min(), -1):
+            names = []
+            for g in self.gb.grids_of_dimension(dim):
+                for e, _ in self.gb.edges_of_node(g):
+                    if isinstance(key[0], tuple):
+                        names += self.variables_of_grid(e)
+            unique_vars = list(set(names))
+            s += f"All variables present on edges between dimensions {dim} and {dim-1}"
+            s += f" {unique_vars}"
+
+            # Also check if some subdomains of this dimension have a subset of the
+            # variables defined on the totality of the subdomains
+
+            # List of found special (subset) variable combinations
+            found_special_var_combination: List[Set[str]] = []
+            for g in self.gb.grids_of_dimension(dim):
+                for e, _ in self.gb.edges_of_node(g):
+                    if isinstance(key[0], tuple):
+                        var = set(self.variables_of_grid(e))
+                        # Check if this is a subset of the full variable list on this
+                        # dimension
+                        if len(var) != len(var.intersection(set(unique_vars))):
+
+                            # We will only report each subset variable definition once.
+                            # By default, assume this subset is new
+                            new_comb = True
+                            # Check if we have found it before
+                            for spec in found_special_var_combination:
+                                if len(spec) == len(spec.intersection(set(var))):
+                                    new_comb = False
+                                    break
+                            # if not bound before, register the new one, and report it
+                            if new_comb:
+                                found_special_var_combination.append(var)
+                                s += "Variable subset on at least one interface between"
+                                s += " dimension {dim} and {dim-1}: {var}\n"
```

Comment:
This change does not use `key` (i.e. element from `self.block_dof.keys()`). I haven't tested properly what that means in practice
```suggestion
            unique_vars = {
                var
                for g in self.gb.grids_of_dimension(dim)
                for e, _ in self.gb.edges_of_node(g)
                for var in self.variables_of_grid(e)
            }

            s += f"All variables present on edges between dimensions {dim} and {dim-1}: " \
                 f"{unique_vars}\n"

            # Also check if some subdomains of this dimension have a subset of the
            # variables defined on the totality of the subdomains

            # List of found special (subset) variable combinations
            found_special_var_combination: List[Set[str]] = []
            for g in self.gb.grids_of_dimension(dim):
                for e, _ in self.gb.edges_of_node(g):
                    var = set(self.variables_of_grid(e))
                    # Check if this is a subset of the full variable list on this dimension
                    if var.issubset(unique_vars):
                        # We will only report each subset variable definition once.
                        # If this subset hasn't already been reported, report it.
                        already_reported = np.any(var == spec for spec in found_special_var_combination)
                        if not already_reported:
                            found_special_var_combination.append(var)
                            s += f"Variable subset on at least one interface between " \
                                 f"dimension {dim} and {dim-1}: {var}\n"
```

Reply:
I slightly updated my suggestion with the code:
```
            unique_vars = {
                var
                for g in self.gb.grids_of_dimension(dim)  # For each grid of dimension dim
                for e, _ in self.gb.edges_of_node(g)      # for each edge of that grid
                if self.gb.nodes_of_edge(e)[1] == g       # such that the edge neighbors a lower-dimensional grid
                for var in self.variables_of_grid(e)      # get all variables on that edge
            }
```
where the fifth line here is new relative to my original suggestion. This ensures that only edges neigboring a lower-dimensional grid relative to the current dimension is considered.